web-wallet/src/lib/containers/Cards/ToggleableCard.svelte

```diff
@@ -26,15 +26,19 @@
   }
 </script>
 
-<Card {...$$restProps} {gap} {onSurface}>
-  <header slot="header" class="dusk-card__header">
-    <div class="dusk-card__header-title">
+<Card {...$$restProps} {gap} {onSurface} showBody={isToggled}>
+  <header slot="header" class="dusk-card__header dusk-card__header-toggle">
+    <h3 class="h4">{heading}</h3>
+    <div class="dusk-card__header-controls-wrapper">
+      <Switch
+        onSurface
+        bind:value={isToggled}
+        on:change={dispatchToggleEvent}
+      />
       {#if iconPath}
         <Icon path={iconPath} />
       {/if}
-      <h3 class="h4">{heading}</h3>
     </div>
-    <Switch onSurface bind:value={isToggled} on:change={dispatchToggleEvent} />
   </header>
   {#if isToggled}
```

Comment:
As the existence of the slot is conditional, I would remove the extra `showBody` prop in the card and just add a condition there to check if the default slot content exists before rendering it.

Having said that: do we really want to have conditional rendering on the card's body instead of simply show / hide it?
Less DOM operations and the show / hide can be also animated.

Reply:
On the first point:

In our case, the default slot is always present. As explained [here](https://stackoverflow.com/questions/78550976/conditional-default-slot-detection-in-svelte):

_Everything inside the component that is not explicitly assigned to a named slot becomes part of the default slot. The {#if} is also part of the slot, so the slot will always be set._

When it comes to _do we really want to have conditional rendering on the card's body instead of simply show / hide it?_ – I am not sure I understand. Do you mean like setting CSS to `display: hidden`, rather than removing/adding it to the DOM?

Reply:
> On the first point:
> 
> In our case, the default slot is always present. As explained [here](https://stackoverflow.com/questions/78550976/conditional-default-slot-detection-in-svelte):

Ok, but what I meant is you should be able to check what `$$slots.default` contains in the child: it should be `undefined`  if the condition is false.
What the guy did in stackoverflow was checking the existence of the slot itself, which is another matter.

> When it comes to _do we really want to have conditional rendering on the card's body instead of simply show / hide it?_ – I am not sure I understand. Do you mean like setting CSS to `display: hidden`, rather than removing/adding it to the DOM?

Yes, maybe along with settings its height / visibility to make it animatable.